### DISCUSSION TITLE
Is there a way to add keymaps based on filetype?

### DISCUSSION THREAD

**Discussion by SymphonySimper** [nix-community/nixvim#2599] [Q&A] [answered]
Currently I'm adding keymaps using `extraFiles."ftplugin/<filetype>"` to enable keymaps based on filetype. Is there a module on nixvim to do this?

**Comment by MattSturgeon** [comment, marked as answer]
One approach would be to use `files.<filename>` instead of `extraFiles`, as that one takes a nixvim module instead of lua content.

Another approach would be to use `keymapsOnEvent` to target a filetype-related autocommand event.

**Reply by SymphonySimper** [reply]
Ohh, noice. Thanks!
